contracts/sponsorship/SponsorshipPaymaster.sol

```diff
@@ -280,6 +280,7 @@ contract SponsorshipPaymaster is
         paymasterIdBalances[feeCollector] += actualPremium;
 
         emit GasBalanceDeducted(paymasterId, costIncludingPremium, userOpHash);
```

Comment:
can add actualPremium in the GasBalanceDeducted event as 4th argument.

> That way we won't need to emit balToDeduct coz that can be calculated using costIncludingPremium-actualPremium

This balToDeduct we are not emitting currently also. but yes can be calculated.

I will make changes in the event in different PR. because any change in contract I need to show to auditors later as diffs. This is meant for accounting test cases so can you please approve and merge it